src/ClientModel/SamplingDataPostProcessorTest.cpp

```diff
@@ -0,0 +1,662 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/container/flat_hash_set.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <vector>
+
+#include "ClientData/ModuleManager.h"
+#include "ClientModel/CaptureData.h"
+#include "ClientModel/SamplingDataPostProcessor.h"
+#include "OrbitBase/ThreadConstants.h"
+#include "capture_data.pb.h"
+
+using orbit_client_data::ModuleManager;
+using orbit_client_data::PostProcessedSamplingData;
+using orbit_client_data::SampledFunction;
+using orbit_client_data::ThreadSampleData;
+
+using orbit_client_protos::CallstackEvent;
+using orbit_client_protos::CallstackInfo;
+using orbit_client_protos::LinuxAddressInfo;
+
+using orbit_grpc_protos::CaptureStarted;
+
+using ::testing::ElementsAre;
+using ::testing::Eq;
+using ::testing::Pointwise;
+using ::testing::UnorderedElementsAre;
+
+namespace orbit_client_model {
+
+namespace {
+
+SampledFunction MakeSampledFunction(std::string name, std::string module_path, float exclusive,
+                                    float inclusive, uint64_t absolute_address) {
+  SampledFunction sampled_function;
+  sampled_function.name = std::move(name);
+  sampled_function.module_path = std::move(module_path);
+  sampled_function.exclusive = exclusive;
+  sampled_function.inclusive = inclusive;
+  sampled_function.absolute_address = absolute_address;
+  sampled_function.function = nullptr;
+  return sampled_function;
+}
+
+bool operator==(const SampledFunction& lhs, const SampledFunction& rhs) {
+  return lhs.name == rhs.name && lhs.module_path == rhs.module_path &&
+         lhs.exclusive == rhs.exclusive && lhs.inclusive == rhs.inclusive &&
+         lhs.absolute_address == rhs.absolute_address && lhs.function == rhs.function;
+}
```

Comment:
Hhm, the reason I haven't done it is just that I'm not sure the type should have a comparison operator. I mean, it can have it, but from the semantics I don't really see why we would want to compare objects of this `struct`, unless it's for testing (and even for testing, it's just for convenience, the logic could be all in the matcher without the `operator==`).

Reply:
pers pref: use `std::tie(lhs.name, ...) == std::tie(rhs.name, ...)`

Reply:
Okay. In that case you could consider defining regular functions instead of operators. Otherwise it's a bit confusing to have them defined in a different namespace.

Reply:
@antonrohr What I don't like about `std::tie` is that while it brings all the `lhs`s and all the `rhs`s together, it moves the matching fields away from each other. (The result from `clang-format` is also somewhat weird, for what it matters.)

@beckerhe That is fair, done (I only left one, move most stuff directly in the matchers).